File: 백준알고리즘/DFS(깊이 우선 탐색)/python/바이러스.py
import sys
input = sys.stdin.readline

N = int(input())
M = int(input())
graph = [[0] * (N + 1) for _ in range(N + 1)]

for _ in range(M):
    m1, m2 = map(int, input().split())
    graph[m1][m2] = graph[m2][m1] = 1

visited = [0 for _ in range(N + 1)]


def dfs(n):
    visited[n] = 1

    for i in range(N + 1):
        if graph[n][i] == 1 and visited[i] == 0:
            dfs(i)


dfs(1)

cnt = 0
for i in visited:
    if i == 1:
        cnt += 1
print(cnt - 1)

File: 백준알고리즘/DFS(깊이 우선 탐색)/javascript/바이러스.js
const fs = require('fs');
const stdin = fs.readFileSync('/dev/stdin').toString().split('\n');
const input = (() => {
  let line = 0;
  return () => stdin[line++];
})();

const N = +input();
const C = +input();

const graph = Array.from(Array(N + 1), () => Array());
const visited = Array.from(Array(N + 1), () => 0);
const answer = [];

for (let i = 0; i < C; i++) {
  const [u, v] = input()
    .split(' ')
    .map(n => +n);

  graph[u].push(v);
  graph[v].push(u);
}

function dfs(node) {
  visited[node] = 1;
  answer.push(node);

  for (const item of graph[node]) {
    if (visited[item] === 0) dfs(item);
  }
}

dfs(1);
console.log(answer.length - 1);
